lib/accessory.js

```diff
@@ -9,34 +9,44 @@ let Service, Characteristic;
 
 class Accessory {
 
-  constructor (raw, overrides) {
+  constructor (raw, overrides, log) {
     assertString(raw.powerstrip_id, '`raw.powerstrip_id` must be set');
     assertString(raw.name, '`raw.name` must be set');
     assertString(raw.device_manufacturer, '`raw.device_manufacturer` must be set');
     assertString(raw.model_name, '`raw.model_name` must be set');
     assertString(raw.serial, '`raw.serial` must be set');
     assertArray(raw.outlets, '`raw.outlets` must be an array');
     assertArray(overrides, "`overrides` must be an array");
-
+	
+	this.log = log;
     this.id = raw.powerstrip_id;
     this.name = raw.name;
     this.manufacturer = raw.device_manufacturer;
     this.model = raw.model_name;
     this.serialNumber = raw.serial;
+	this.log("Strip " + this.name + " ('" + this.model + " - S/N '" + this.serialNumber + "') initialized.");
+	
+	var that = this;
+	
     this.outlets = raw.outlets.map(outlet => {
       const override = this.outlet(overrides, outlet.uuid) || {};
       const outletConfig = Object.assign({
         uuid: outlet.uuid,
         name: outlet.name,
-        hapService: 'Outlet'
+        hapService: 'Outlet',
+		log : this.log
       }, override);
       assert(Service[outletConfig.hapService] !== undefined, `${outletConfig.hapService} is not a known HAP service.`);
       return outletConfig
     });
   }
 
   outlet (outlets, uuid) {
-    return outlets.filter(outlet => outlet.uuid === uuid)[0];
+    if(outlets && outlets.filter){
```

Comment:
Couple things here:
- `outlets` should always be defined (it comes from the `config.json`, if `undefined` we should probably have an assert statement upfront when instantiating the plugin - but for now let's assume it's defined - and `filter` property of `outlets` should also exists since `outlets` _should_ be an array (per config requirement). So no need for checking `outlets && outlets.filter`.
- we can use the [es6 `find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) method in place of `filter`, e.g
`return  outlet.find(outlet => outlet.uuid === uuid)`. If no outlet found, it'll return `undefined` without crashing.